src/ert/storage/local_ensemble.py

```diff
@@ -357,15 +353,24 @@ def load_gen_data_as_df(
         return pd.concat(dfs)
 
     def load_field(self, key: str, realizations: List[int]) -> npt.NDArray[np.double]:
-        result = []
+        result: Optional[npt.NDArray[np.double]] = None
         for realization in realizations:
             input_path = self.mount_point / f"realization-{realization}"
             if not input_path.exists():
                 raise KeyError(f"Unable to load FIELD for key: {key}")
             data = np.load(input_path / f"{key}.npy")
-            data_no_nans = data[~np.isnan(data)]
-            result.append(data_no_nans)
-        return np.stack(result).T  # type: ignore
+            data = data[np.isfinite(data)]
```

Comment:
What if we try using memory-mapping here?
So, I have not tried this but the docs of `np.load` state that:

_mmap_mode{None, ‘r+’, ‘r’, ‘w+’, ‘c’}, optional
If not None, then memory-map the file, using the given mode (see [numpy.memmap](https://numpy.org/doc/stable/reference/generated/numpy.memmap.html#numpy.memmap) for a detailed description of the modes). A memory-mapped array is kept on disk. However, it can be accessed and sliced like any ndarray. Memory mapping is especially useful for accessing small fragments of large files without reading the entire file into memory._

So something like this should work I think:

```python
# Memory-map the .npy file
data = np.load(input_path / f"{key}.npy", mmap_mode='r')
# Create a boolean mask for finite values
finite_mask = np.isfinite(data)
# Load only finite values
finite_data = data[finite_mask]
```

Reply:
This was added